### DISCUSSION TITLE
为什么组件用渲染函数来写？

### DISCUSSION THREAD

**Discussion by Beny2305** [vuepress-theme-hope/vuepress-theme-hope#3286] [General]
用渲染函数开发的组件，虽然组件不是很复杂，作者开发的时候也许很方便，但在覆盖组件时添加元素比较复杂的维护难度高，最后还是要改成模版方式。有没有觉得要改成模版方式，会更加有利于推广？

**Comment by Mister-Hope** [comment]
没有，模板冷启动会慢一些，而且打包很困难。

另外纯动态组件使用渲染函数的维护性要远高于模板。

**Reply by Beny2305** [reply]
行吧，对开发场景来说你这种方式确实精简。但是想要添加一个vue的第三方组件库，就很不方便了，我就这么一说，你可以考虑考虑
